aibolit/__main__.py

```diff
@@ -327,6 +328,23 @@ def run_recommend_for_file(file: str, args):
     :return: dict with code lines, filename and pattern name
     """
     java_file = str(Path(os.getcwd(), file))
+    tree = build_ast(file)
+    classes_with_annonations = find_annotation_by_node_type(tree, javalang.tree.ClassDeclaration)
+    functions_with_annotations = find_annotation_by_node_type(tree, javalang.tree.MethodDeclaration)
+    fields_with_annotations = find_annotation_by_node_type(tree, javalang.tree.FieldDeclaration)
+    classes_with_patterns_ignored = flatten(
+        [pattern_code for node, pattern_code in classes_with_annonations.items()])
+    patterns_ignored = defaultdict(list)
+
+    for node, patterns_list in functions_with_annotations.items():
+        start_pos, end_pos = find_start_and_end_lines(node)
+        for p in patterns_list:
+            patterns_ignored[p].append([start_pos, end_pos])
+
+    for node, patterns_list in fields_with_annotations.items():
+        for p in patterns_list:
+            patterns_ignored[p].append([node.position.line, node.position.line])
```

Comment:
Why duplicates?

Reply:
because `field declaration`  is  usually 1 line. And we do not find start and end position for it.